SmartPlaces.Facilities/samples/Telemetry.Mapped/src/Program.cs

```diff
@@ -0,0 +1,96 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="Program.cs" company="Microsoft">
+//   Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Telemetry
+{
+    using System;
+    using System.Threading.Tasks;
+    using Azure.Identity;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.SmartPlaces.Facilities.IngestionManager;
+    using Microsoft.SmartPlaces.Facilities.IngestionManager.Extensions;
+    using Microsoft.SmartPlaces.Facilities.IngestionManager.Interfaces;
+    using Telemetry.Interfaces;
+    using Telemetry.Processors;
+
+    public class Program
+    {
+        /// <summary>
+        /// Entry point into Telemetry
+        /// </summary>
+        /// <param name="args">An option for passing in configuration values, though not required</param>
+        /// <exception cref="ArgumentNullException">Thrown when required configuration values are not present</exception>
+        public static async Task Main(string[] args)
+        {
+            Console.WriteLine("Starting Telemetry Processing");
+
+            using IHost host = Host
+                .CreateDefaultBuilder(args)
+                .ConfigureAppConfiguration((hostContext, config) =>
+                {
+                    var settings = config.Build();
+
+                    var vaultUri = settings["KeyVaultEndpoint"];
+
+                    if (string.IsNullOrWhiteSpace(vaultUri))
+                    {
+                        Console.WriteLine("Startup Variable 'KeyVaultEndpoint' not found. Exiting");
+                        throw new ArgumentNullException("KeyVaultEndpoint");
+                    }
+
+                    config.AddAzureKeyVault(new Uri(vaultUri), new DefaultAzureCredential());
+                })
+                .ConfigureServices((hostContext, services) =>
+                {
+                    // Configure ILogger
+                    services.AddLogging();
+
+                    // Configure options for ApplicationInsights
+                    services.AddApplicationInsightsTelemetryWorkerService(options =>
+                    {
+                        options.ConnectionString = hostContext.Configuration["AppInsightsConnectionString"];
+                        options.EnableAdaptiveSampling = false;
+                    });
+
+                    // This is for getting the TwinId and TwinModelId 
+                    // based on the MappingKey which is provided by Mapped.
+                    // Implements ITwinMappingIndexer, IOutputGraphManager
```

Comment:
It seems services.AddIngestionManager, in spite of the name, only adds IOutputGraphManger; which in turn requires but does not implement an ITwinMappingIndexer.

Reply:
Good catch. It used to, however, was removed for the scenario where a customer doesn't have a Cloud Redis cache because they dont ingest the telemetry into ADT and use the externalId to join telemetry later in their pipeline. I think we can re-add it with a flag, but would consider that work out of scope for this PR.